chcemvediet/tests/__init__.py

```diff
@@ -20,3 +30,124 @@ def setup_test_environment(self, **kwargs):
     def run_tests(self, *args, **kwargs):
         logging.disable(logging.CRITICAL)
         return super(CustomTestRunner, self).run_tests(*args, **kwargs)
+
+class ChcemvedietTestCaseMixin(TestCase):
+
+    def _pre_setup(self):
+        super(ChcemvedietTestCaseMixin, self)._pre_setup()
+        self.user = self._create_user()
+        self.region = self._create_region(id=u'SK00000', name=u'SK00000')
+        self.district = self._create_district(id=u'SK00000', name=u'SK00000')
+        self.municipality = self._create_municipality(id=u'SK00000', name=u'SK00000')
+        self.neighbourhood = self._create_neighbourhood(id=u'SK00000', name=u'SK00000')
+        self.inforequest = self._create_inforequest()
+        self.obligee = self._create_obligee(name=u'ChcemvedietTestCase Obligee')
```

Comment:
@viliambalaz Je nejaky dovod, preco tu posielas `name` a nenechas `_create_obligee` pouzit defaultny `name`?

Reply:
Problem je, ze v `_pre_setup` sme vytvorili init data, ktore su dostupne pri spustani jednotlivych testoch. Cize po zavolani 2-krat metody `_create_obligee` (bez paramentru `name`) nastane `IntegrityError: UNIQUE constraint failed:`. Data vytvorene lokalne v teste zaniknu hned po jeho ukonceni. Data vytvorene v `ChcemvedietTestCaseMixin` zaniknu az po skonceni vsetkych testoch v module.

Reply:
@viliambalaz Aha, potom by bolo lepsie, priamo v `_create_obligee` automaticky vytvarat instancie s unikatnym defaultnym `name`.

Napriklad tak, ze si niekde v `_pre_setup` (alebo globalne v classe / module ak to nezafunguje v `_pre_setup`) definujeme nekonecny counter: `self.counter = itertools.count()`. A potom v `_create_obligee` puzijeme `u'name': u'Default Testing Name {}'.format(self.counter.next())`.

Analogicky pre ostatne fieldy, ktore musia byt unikatne. Napriklad aj pre stringy ako `SK00000` je lepsie pouzit `u'SK{:05d}'.format(self.counter.next())` namiesto random hodnot. Kedze pri random hodnotach raz za cas vznikne kolizia. Pri country kolizia nevznikne nikdy.

Pre rozne fieldy mozeme mat nezavisle countre, alebo mozeme mat jeden spolocny counter. Ako ti je lepsie.

Reply:
:ok:

Reply:
Pouzil som `{:05d}` format aj pre `Obligee.name`, aby nazvy vytvaranych instancii sa dal usporiadat za sebou podla ich vytvorenia.

Reply:
@viliambalaz V principe OK. Ale neni mi jasne, kde ich treba usporiadavat?

Reply:
Bolo to myslene pre `test_order_by_name`. Teraz si uvedomujem ze build-in `sorted()` sa bude spravat rovnako pri usporaduvani prvkov. Takze nebudem pridavat nuly a zbytocne  to komplikovat, kedze to netreba. dakujem :ok:

Reply:
@viliambalaz V `test_order_by_name_query_method` predsa aj tak sortime iba tie obligees, ktore vytvorime v teste a dame im vlastny `name`, nie defaultny.

In general v testoch by sme mali vzdy porovnavat iba hodnoty, ktore v tychtestoch explicitne nastavime, nie defaultne hodnoty. Teda, ze testy by mali fungovat aj ak by v `_create_obligee` sme pouzivali random default hodnoty.

Reply:
@martinmacko47 Po pridani iteratora na definovanie nazvov, som v testoch odstranil parameter `name` tam, kde bol nadbytocny.